lib/process_balancer/manager.rb

```diff
@@ -117,11 +117,26 @@ def update_state
       [exists, msg]
     end
 
+    def stale_workers(connection, workers)
+      workers.reject { |i| connection.exists?(i) }
+    end
+
     def update_process_index
       redis do |c|
         c.watch(PROCESSES_KEY)
 
-        workers     = c.lrange(PROCESSES_KEY, 0, -1)
+        workers   = c.lrange(PROCESSES_KEY, 0, -1)
+        stale_ids = workers.reject { |i| c.exists(i) }
```

Comment:
WDYT if we get rid of the 2nd fetch of `workers` on line `138` by changing this line to:
```ruby
workers, stale_ids = workers.reject { |i| c.exists(i) }
```

Reply:
Except that I am setting up a transaction there with the "watch" so I really need to fetch it after I watch otherwise things could be out of sync.